.github/workflows/unit-nvidia-reusable.yaml

```
@@ -0,0 +1,132 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: "Run unit tests on specified EC2"
+
+on:
+  workflow_call:
+    inputs:
+      ec2_runner_variant:
+        description: "EC2 instance type of runner"
+        type: string
+        default: "g6e.12xlarge" # TODO: consider smaller default instance.
+      pytest_mark:
+        type: string
+        description: "pytest.mark of tests that will be run"
+        default: "fast"
+
+jobs:
+  start-ec2-runner:
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.label }}
+
+    steps:
+      - name: "Harden runner"
+        # v2.10.1
+        uses: step-security/harden-runner@0080882f6c36860b6ba35c610c98ce87d4e2f26f
```

```suggestion
        uses: step-security/harden-runner@0080882f6c36860b6ba35c610c98ce87d4e2f26f # v2.10.1
```

Put the comment on the same line. Dependabot will mod the comment when there are new versions.

---

done